airflow/timetables/simple.py

```diff
@@ -55,6 +55,9 @@ def serialize(self) -> dict[str, Any]:
     def infer_manual_data_interval(self, *, run_after: DateTime) -> DataInterval:
         return DataInterval.exact(run_after)
 
+    def validate(self):
```

Comment:
I'm not quite sure how these time tables were working. Mypy complained about creating an abstract class anyway (i guess it's a difference in what it thinks Protocols do vs how they work at runtime?)

PTAL at this change @uranusjr

Reply:
TIL.  Empty methods - whether ... or just with docstring **just work**. No matter if they are part of protocol or not:

This one works fine and exit with 0 error code:

```python
from typing_extensions import Protocol


class A(Protocol):
    def method(self) -> None:
        """Empty method"""

    def method2(self) -> None:
        ...


class B(A):
    def method3(self) -> None:
        ...


B().method()
B().method2()
B().method3()
```

Reply:
I suspect we are "unusual" in inheriting from a Protocol (we are treating it somewhat like an ABC) and I guess mypy is somehow ignoring the methods on here.

Reply:
PEP 544 has examples about subclassing a base protocol and paragraphs talking specifically about default implementations, so those should not be special in any way. Definitely a bug in Mypy.

Reply:
:eyes: instead of :man_facepalming:

Reply:
I kind of figured out what’s going on. Mypy seems to treat a function with `pass` as “empty” and ignore its presence. It’s a bug, but easier to work around (by replacing `pass` with `return`).